govcd/api.go

```diff
@@ -127,6 +129,12 @@ func ContainsNotFound(err error) bool {
 
 // NewRequestWitNotEncodedParams allows passing complex values params that shouldn't be encoded like for queries. e.g. /query?filter=name=foo
 func (cli *Client) NewRequestWitNotEncodedParams(params map[string]string, notEncodedParams map[string]string, method string, reqUrl url.URL, body io.Reader) *http.Request {
+	return cli.NewRequestWitNotEncodedParamsWithApiVersion(params, notEncodedParams, method, reqUrl, body, cli.APIVersion)
```

Comment:
I'd like to see a few tests for the new functions, to make sure that:
1. They are really using the intended API version.
2. Subsequent requests using the regular `Execute*` methods are not affected.

Reply:
Could you share your idea how to test that?

Reply:
Here's a possible way:
1. Add fields to a type that was enhanced in API versions later than the current one.
2. Run a GET that retrieves the type with the enhanced fields
3. Make sure the fields are non-empty
4. Run the same query without the new API version
5. Make sure that the enhanced fields are empty

Reply:
Two more ideas:
* https://golang.org/pkg/net/http/httptest/
or 
* IIRC when you send a request on a specific version, vCD responds with API version (in headers). You could probably check that.

Reply:
first option could be valid with on specific vCD version or I have to find valid for all versions, which is lottery
I'll check possibility with response headers

Reply:
Test added